nibabel/streamlines/tests/test_array_sequence.py

```diff
@@ -277,6 +277,128 @@ def test_arraysequence_getitem(self):
+            seq = getattr(SEQ_DATA['seq'], op)(SEQ_DATA['seq'])
+            assert_true(seq is not SEQ_DATA['seq'])
+            check_arr_seq(seq, [getattr(d, op)(d) for d in SEQ_DATA['data']])
+
+            # Test math operators with ArraySequence views.
+            orig = SEQ_DATA['seq'][::2]
+            seq = getattr(orig, op)(orig)
+            assert_true(seq is not orig)
+            check_arr_seq(seq, [getattr(d, op)(d) for d in SEQ_DATA['data'][::2]])
+
+        # Test in-place operators.
+        for op in ["__iadd__", "__isub__", "__imul__", "__ifloordiv__", "__itruediv__"]:
+            # Test in-place math operators with a scalar.
+            for scalar in [42, 0.5, True]:
+                seq = seq_orig = SEQ_DATA['seq'].copy()
+                seq = getattr(seq, op)(scalar)
+                assert_true(seq is seq_orig)
+                check_arr_seq(seq, [getattr(d.copy(), op)(scalar) for d in SEQ_DATA['data']])
+
+            # Test in-place math operators with another ArraySequence.
+            seq = seq_orig = SEQ_DATA['seq'].copy()
+            seq = getattr(seq, op)(SEQ_DATA['seq'])
+            assert_true(seq is seq_orig)
+            check_arr_seq(seq, [getattr(d.copy(), op)(d) for d in SEQ_DATA['data']])
+
+            # Test in-place math operators with ArraySequence views.
+            seq = seq_orig = SEQ_DATA['seq'].copy()[::2]
+            seq = getattr(seq, op)(seq)
+            assert_true(seq is seq_orig)
+            check_arr_seq(seq, [getattr(d.copy(), op)(d) for d in SEQ_DATA['data'][::2]])
+
+            # Operations between array sequences of different lengths.
+            seq = SEQ_DATA['seq'].copy()
+            assert_raises(ValueError, getattr(seq, op), SEQ_DATA['seq'][::2])
+
+            # Operations between array sequences with different amount of data.
+            seq1 = ArraySequence(np.arange(10).reshape(5, 2))
+            seq2 = ArraySequence(np.arange(15).reshape(5, 3))
+            assert_raises(ValueError, getattr(seq1, op), seq2)
+
+            # Operations between array sequences with different common shape.
+            seq1 = ArraySequence(np.arange(12).reshape(2, 2, 3))
+            seq2 = ArraySequence(np.arange(8).reshape(2, 2, 2))
+            assert_raises(ValueError, getattr(seq1, op), seq2)
+
+        # Unary operators
+        for op in ["__neg__"]:
+            seq = getattr(SEQ_DATA['seq'], op)()
+            assert_true(seq is not SEQ_DATA['seq'])
+            check_arr_seq(seq, [getattr(d, op)() for d in SEQ_DATA['data']])
+
+    def test_arraysequence_setitem(self):
+        # Set one item
+        seq = SEQ_DATA['seq'] * 0
+        for i, e in enumerate(SEQ_DATA['seq']):
+            seq[i] = e
+
+        check_arr_seq(seq, SEQ_DATA['seq'])
+
+        if sys.version_info < (3,):
+            seq = ArraySequence(SEQ_DATA['seq'] * 0)
+            for i, e in enumerate(SEQ_DATA['seq']):
+                seq[long(i)] = e
```

Comment:
Nibabel 3.0 will be Python 3-only. You can drop this check.

Reply:
Will do.